### PR TITLE
Dangermattic checks update

### PR DESCRIPTION
**Pull request by iangmaia**
This PR adapts the `Dangerfile` to the latest [Dangermattic](https://github.com/Automattic/dangermattic) version, using newly added plugins, adapting the code as needed and implementing a few improvements.

The main changes:
- Improvements to the Rubocop check call
- Added Core Data model changed check
- Added Podfile branch references check
- Added Release Notes / Store strings check
- Added Tracks Check
- Added check for translations being changed only on the release branch
- Added check on the internal release notes being changed 
- Use `github_utils` plugin instead of locally declared functions
- Added message report when we're skipping checks due to the `Releases` label
- Use officially distributed `danger-dangermattic` Gem

## Testing
The main validation is to make sure CI is 🟢.

To run Danger yourself:
1. Checkout this branch 
2. Run `bundle install`
3. Run:
```ruby
DANGER_GITHUB_API_TOKEN=<github_token> bundle exec danger pr <PR_URL>
```
You should get in the console the same errors / warnings reported in the PR.

### CODE REVIEW

**Comment by dangermattic** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">PR is not assigned to a milestone.</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_pr-check">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

File: Dangerfile [hunk 1]
```diff
@@ -1,37 +1,49 @@
...
+rubocop.lint(files: [], force_exclusion: true, inline_comment: true, fail_on_inline_comment: true, include_cop_names: true)
+
+manifest_pr_checker.check_all_manifest_lock_updated
+
+podfile_checker.check_podfile_does_not_have_branch_references
 
-def wip_feature?
-  has_wip_label = github.pr_labels.any? { |label| label.include?('WIP') }
-  has_wip_title = github.pr_title.include?('WIP')
+ios_release_checker.check_core_data_model_changed
```
Comment (mokagio) [inline Dangerfile:12]: Dangermattic-level note/nitpick. Apologies for not noticing this earlier, but seeing the method in action I do wonder about the naming.

"Check Core Data Model changed" makes me think that we expect the model to have changed. Whereas what we are trying to do is ensure that _if_ the model changed, it happened only before the build has been released for beta testing yet, https://github.com/Automattic/dangermattic/blob/37da73a5d70f2009868a6e4d83d9a4003e9d96c2/lib/dangermattic/plugins/ios_release_checker.rb#L25-L40.

I don't have a strong suggestion for a name that would make this clearer... `alert_on_core_data_model_changed` maybe? I considered `warn_on...` but the type of message is configurable.

File: Dangerfile [hunk 2]
```diff
@@ -1,37 +1,49 @@
...
 
-view_changes_need_screenshots.view_changes_need_screenshots
-
-pr_size_checker.check_diff_size
-
 # skip check for draft PRs and for WIP features unless the PR is against the main branch or release branch
-milestone_checker.check_milestone_due_date(days_before_due: 2) unless github.pr_draft? || (wip_feature? && !(release_branch? || main_branch?))
-
-rubocop.lint(inline_comment: true, fail_on_inline_comment: true, include_cop_names: true)
+milestone_checker.check_milestone_due_date(days_before_due: 2) unless github_utils.wip_feature? && !(github_utils.release_branch? || github_utils.main_branch?)
```
Comment (mokagio) [inline Dangerfile]: A nitpick for your consideration. 

```suggestion
milestone_checker.check_milestone_due_date(days_before_due: 2) unless github_utils.wip_feature? && !github_utils.release_branch? && !github_utils.main_branch?
```

Or, here's another option. What we want to do is check the milestone if we are targeting the release branch or the main branch and we are not on a WIP feature, so

```suggestion
milestone_checker.check_milestone_due_date(days_before_due: 2) if (github_utils.main_branch? || github_utils.release_branch?) && !github_utils.wip_feature?
```

**Review by mokagio — APPROVED** [review]
Approving to unblock. The same comments from the Pocket Casts iOS PR apply here, https://github.com/Automattic/pocket-casts-ios/pull/1418#pullrequestreview-1876795337, but none of them were blocking.

File: Dangerfile (hunk 1 above)
Comment (mokagio) [reply Dangerfile:12]: Or, like `ios_release_checker.check_release_notes_and_app_store_strings` in the line below, a simpler

```ruby
ios_release_checker.check_core_data_model
```

Comment (iangmaia) [reply Dangerfile:12]: Good suggestion -- I've [created an issue](https://github.com/Automattic/dangermattic/issues/56) for it.

**Pull request merged**